core/src/main/java/org/elasticsearch/transport/TransportMessage.java

```diff
@@ -22,14 +22,18 @@
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.transport.TransportAddress;
 
 import java.io.IOException;
 
-public abstract class TransportMessage implements Streamable {
+public abstract class TransportMessage implements Streamable, Writeable {
 
     private TransportAddress remoteAddress;
 
+    public TransportMessage() {
```

Comment:
a single, explicit empty ctor should not be necessary (that is what the default ctor is)